app/services/customers/update_invoice_custom_sections_service.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Customers
+  class UpdateInvoiceCustomSectionsService < BaseService
+    def initialize(customer:, section_ids: [])
+      @customer = customer
+      @section_ids = section_ids
+
+      super
+    end
+
+    def call
+      return result.not_found_failure!(resource: "customer") unless customer
+      return result if customer.applicable_invoice_custom_sections.ids == section_ids
+
+      if organization.selected_invoice_custom_sections.ids == section_ids
+        assign_organization_sections
+      else
+        assign_customer_sections
+      end
+      result
+    rescue ActiveRecord::RecordInvalid => e
+      result.record_validation_failure!(record: e.record)
+    end
+
+    private
+
+    attr_reader :customer, :section_ids
+
+    def organization
+      @organization ||= customer.organization
+    end
+
+    def assign_organization_sections
+      # Note: when inheriting organization's selections, customer shouldn't have their selected sections
+      customer.selected_invoice_custom_sections = []
+    end
```

Comment:
What about if the customer selected custom sections are the same as the organization by coincidence but the user expect to not change them if the organization one changes? This use case is not feasible with this implementation.

What about having a flag to know when the customer should inherit from organization or have a custom selection?

Reply:
you're absolutely right. Actually further with the FE we agreed to send empty array when customer follows organization's selection, forgot about this piece. thank you!